src/analysis/brute-force.ts

```diff
@@ -156,3 +143,32 @@ export default async function analyzeBruteForce(directory: string, reference: st
     }
   }
 }
+
+/**
+ * Prepares the resolution-bitrate pairs to be analyzed.
+ *
+ * @param resolutions The different resolutions to be analyzed. Also contains the bitrate range.
+ * @param bitrates A list of bitrates that can analyzed.
+ * @param filterFunction A filter function to be used for filtering out bitrates.
+ */
+export function preparePairs(resolutions: Resolution[], bitrates: number[], filterFunction: (pair: BitrateResolutionPair) => boolean): BitrateResolutionPair[] {
+  return resolutions
+    .flat()
+    .flatMap(resolution =>
+      bitrates.map(bitrate =>
+        (filterFunction({ bitrate, resolution }) && (resolution.range === undefined || checkIfBitrateInRange(bitrate, resolution.range))) 
+          ? {
+              resolution,
+              bitrate,
+            }
+          : undefined
+      )
+    )
+    .filter(pair => pair !== undefined) as BitrateResolutionPair[];
+}
+
+function checkIfBitrateInRange(bitrate: number, range: BitrateRange): boolean {
+  let minBitrate = range?.min != undefined ? range.min : defaultBitrates[0];
+  let maxBitrate = range?.max != undefined ? range.max : defaultBitrates[defaultBitrates.length - 1];
```

Comment:
`!= undefined` confuses me more than it should 😅 and maybe that's just me. It can safely be removed but ultimately it does not matter.

Reply:
I didn't know you could skip the condition part! 😅
Fixed!